src/ol/layer/tilelayer.js

```diff
@@ -29,14 +30,16 @@ ol.layer.TileProperty = {
  */
 ol.layer.Tile = function(opt_options) {
   var options = goog.isDef(opt_options) ? opt_options : {};
-  goog.base(this,  /** @type {olx.layer.LayerOptions} */ (options));
-
-  if (!goog.isDef(this.getPreload())) {
-    this.setPreload(0);
-  }
-  if (!goog.isDef(this.getUseInterimTilesOnError())) {
-    this.setUseInterimTilesOnError(true);
-  }
+
+  var baseOptions = goog.object.clone(options);
+
+  delete baseOptions.preload;
+  delete baseOptions.useInterimTilesOnError;
+  goog.base(this,  /** @type {olx.layer.LayerOptions} */ (baseOptions));
+
+  this.setPreload(goog.isDef(options.preload) ? options.preload : 0);
+  this.setUseInterimTilesOnError(goog.isDef(options.useInterimTilesOnError) ?
+      options.useInterimTilesOnError : true);
```

Comment:
> You code works but I suggest to keep the goog.object.clone: it is how it's done in the other classes (ol.layer.Group, ol.layer.Vector, ...)

Ok.



> and we will had to change if if/when we have a new non primitive type to handle.

We'd need to add `delete` statements anyway, even if `clone` is used. No?